src/helpers.php

```diff
@@ -1,10 +1,21 @@
 <?php
 
 use CyrildeWit\EloquentViewable\Views;
+use CyrildeWit\EloquentViewable\Contracts\Viewable as ViewableContract;
 
 if (! function_exists('views')) {
     function views($viewable = null)
     {
-        return app(Views::class)->forViewable($viewable);
+        $builder = app(Views::class);
+
+        if (is_string($viewable)) {
+            $model = app($viewable);
+
+            if ($model instanceof ViewableContract) {
```

Comment:
What about making this a negative condition and throw an InvalidViewable Exception.